Tests/FluentAssertions.Equivalency.Specs/SelectionRulesSpecs.cs

```diff
@@ -1489,5 +1490,89 @@ public void Including_an_interface_property_through_inheritance_should_work()
                 .Including(a => a.Value2)
                 .RespectingRuntimeTypes());
         }
+
+        [Theory]
+        [MemberData(nameof(TestMembers))]
```

Comment:
🤔 I found it very hard to understand what this is doing with the parameter. Because of that, I strongly believe it is much better to drop the parameterized tests and have individual tests for the relevant test cases.

Reply:
Oh? This is just how Xunit works. Maybe this can be improved with comments and/or better naming??

Reply:
I strongly disagree. Parameterized tests are only useful if you have a simple list of values that you use in combination with a `Theory` attribute. For anything else, they obscure the cause and effect of the test. In this case, having 6 different tests would be much clearer.

Reply:
Oh dear :-P Which way should I take it?

Reply:
I find the difference between having primitives in attributes and _my example_ to be insignificant. 
If the test case generator is named something with "browsable" I even think it has benefits as the test names don't have to be cluttered with implementation details such as "always" and "advanced". 
So I think dynamic test cases have their moments, but certainly many pitfalls that harm "readability". 

Dennis and don't agree on everything, but (surprisingly?) on quite many things. 
In the end Dennis has the final call.

Reply:
> What about having a test method that isn't marked [Test] and 6 "head" methods that call the test implementation with different parameters?

I think it would improve the situation a bit. But in the end, I would just spell out the entire test case. Trying to achieve DRY in unit testing usually backfires.

Reply:
In this particular case, though, these really are variations on the _same test_. That's why I personally went for the parameterized approach in the first place. If the nature of the test ever changes, it's guaranteed to need the same edit to every repeated copy of the test. I agree that DRY is _in general_ not a good principle in unit tests, but there are cases like this where I think it makes sense.

Reply:
Interesting 🙃 I prefer the parameterized one to this, as I don't have to jump between method bodies.

Conditionals in tests is a (99.9%) no-go for me. 
For parameterized tests I would create separate test case generators to avoid that.

Seems Dennis and I can agree on spelling out the entire test cases.

Reply:
Okay, conditionals are gone. :-)

Reply:
Sorry, was trying to understand each test, but still had to dig through the private methods to understand what each test was doing. I really want them to be spelled out and have functional names.

Reply:
I have pushed a commit that gives the methods in question much more explicit names. Is that what you had in mind? :-)

Reply:
Unfortunately not. In fact, it makes it worse since you refer to the literal names of the APIs, which is an [anti-pattern for me](https://www.continuousimprover.com/2021/10/laws-test-driven-development.html). What I meant is to drop the parameterized tests completely.